helpers/limit/limits.go

```diff
@@ -26,14 +32,48 @@ func GetMaxGoRoutines() uint64 {
 	return ulimit.Max
 }
 
+// DiagnoseLimits verifies if user should increase ulimit or max file descriptors to improve number of expected
+// goroutines in CQ to improve performance
+func DiagnoseLimits() (diags diag.Diagnostics) {
+	// the amount of goroutines we want based on machine memory
+	want := calculateGoRoutines(getMemory())
+	// calculate file descriptor limit
+	fds, err := calculateFileLimit()
+	if err == nil && fds < want {
+		diags = diags.Add(diag.NewBaseError(errors.New("available file descriptors capacity lower than expected"),
```

Comment:
since it's a "consider" this should be `diag.WARNING`? Same for the one below too

Reply:
correct! my bad, even the one below should be warning

Reply:
fixed